src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -117,15 +117,44 @@ public void analyse(Project project, SensorContext context) {
   }
 
   private void analyze(boolean includeRules) {
+    String analysisSettings = analysisSettings(true, settings.getBoolean("sonar.cs.ignoreHeaderComments"), includeRules, ruleProfile, filesToAnalyze());
+
+    File analysisInput = toolInput();
+    File analysisOutput = toolOutput();
+
+    try {
+      Files.write(analysisSettings, analysisInput, Charsets.UTF_8);
+    } catch (IOException e) {
+      throw Throwables.propagate(e);
+    }
+
+    File executableFile = extractor.executableFile();
+
+    Command command = Command.create(executableFile.getAbsolutePath())
+      .addArgument(analysisInput.getAbsolutePath())
+      .addArgument(analysisOutput.getAbsolutePath());
+
+    int exitCode = CommandExecutor.create().execute(command, new LogInfoStreamConsumer(), new LogErrorStreamConsumer(), Integer.MAX_VALUE);
+    if (exitCode != 0) {
+      throw new IllegalStateException("The .NET analyzer failed with exit code: " + exitCode + " - Verify that the .NET Framework version 4.5.2 at least is installed.");
+    }
+  }
+
+  public static String analysisSettings(boolean includeSettings, boolean ignoreHeaderComments, boolean includeRules, RulesProfile ruleProfile, Iterable<File> files) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Provide multiple methods instead of using "includeSettings" to determine which action to take. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2301)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Provide multiple methods instead of using "includeRules" to determine which action to take. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2301)